util/kubeconfig/testing.go

```diff
@@ -17,20 +17,14 @@ limitations under the License.
 package kubeconfig
 
 import (
-	"context"
 	"fmt"
 
 	"k8s.io/client-go/rest"
 	"k8s.io/client-go/tools/clientcmd"
 	"k8s.io/client-go/tools/clientcmd/api"
 	clusterv1 "sigs.k8s.io/cluster-api/api/v1alpha3"
-	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
-func CreateEnvTestSecret(client client.Client, cfg *rest.Config, cluster *clusterv1.Cluster) error {
-	return client.Create(context.TODO(), GenerateSecret(cluster, FromEnvTestConfig(cfg, cluster)))
-}
```

Comment:
This should be either marked as deprecated and be left here and removed in the next version, we can't have breaking API changes in our